### PR TITLE
[Merged by Bors] - Added transparency to window builder

### LINKED ISSUE #3032: Allow window to be transparent
## What problem does this solve or what need does it fill?

For an example this feature would help users build desktop widgets or something similar to this game: https://samperson.itch.io/desktop-goose

## What solution would you like?

A way to set the transparency of the window on application init or in a system

## What alternative(s) have you considered?

I've tried accessing the `ResMut<Windows>` but no api is exposed to modify the winit window transparency 

## Additional context

It seems that winit allows the transparency to be set via the WindowBuilder but not after the creation of the window.
winit docs:
https://docs.rs/winit/0.25.0/winit/window/struct.WindowBuilder.html#method.with_transparent
https://docs.rs/winit/0.25.0/winit/window/struct.WindowAttributes.html#structfield.transparent

### PR DESCRIPTION
**Pull request by louisgjohnson**
Applogies, had to recreate this pr because of branching issue.
Old PR: https://github.com/bevyengine/bevy/pull/3033

# Objective

Fixes #3032

Allowing a user to create a transparent window 

## Solution

I've allowed the transparent bool to be passed to the winit window builder

### CODE REVIEW

**Comment by MiniaczQ** [comment]
Code looks fine,
sucks that it's not supported on so many platforms.

**Comment by louisgjohnson** [comment]
> Code looks fine, sucks that it's not supported on so many platforms.

Well it only makes sense for Desktop platforms(Windows, Mac, Linux) and:

> I think I found a workaround on Linux, forcing `winit` to run on it's master branch fixes this, or at least just on my Manjaro Linux machine with Nvidia proprietary drivers. I'm assuming that this now works because of this PR: [rust-windowing/winit#2006](https://github.com/rust-windowing/winit/pull/2006)

_Originally posted by @Supermath101 in https://github.com/bevyengine/bevy/issues/3032#issuecomment-954148213_

So if that gets updated in the future then only mac will not be supported which might get fixed by this issue: https://github.com/gfx-rs/wgpu/issues/687

**Review by alice-i-cecile — APPROVED** [review]

**Review by IceSentry — APPROVED** [review]

**Comment by louisgjohnson** [comment]
bors try

**Comment by alice-i-cecile** [comment]
@louisgjohnson, you need to be on the Bevy org to be able to do that :) It's a low bar (pretty much just "good faith"), just ping cart on Discord or Github.

**Comment by alice-i-cecile** [comment]
bors try

**Comment by mockersf** [comment]
> > I think I found a workaround on Linux, forcing `winit` to run on it's master branch fixes this, or at least just on my Manjaro Linux machine with Nvidia proprietary drivers. I'm assuming that this now works because of this PR: [rust-windowing/winit#2006](https://github.com/rust-windowing/winit/pull/2006)
> 
> _Originally posted by @Supermath101 in [#3032 (comment)](https://github.com/bevyengine/bevy/issues/3032#issuecomment-954148213)_

If it won't work by default on linux, then linux should be mentioned in the platform specific documentation

File: examples/window/transparent_window.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/// An example of how to display a window in transparent mode
+/// [Documentation & Platform support.](https://github.com/bevyengine/bevy/blob/latest/crates/bevy_window/src/window.rs)
```
Comment (mockersf) [inline examples/window/transparent_window.rs:2]: Linking to the complete code file can be hard to follow. You could link to the exact line, but that could be a pain to keep track of changes.

Other possibility which I would prefer is to link to docs.rs. The field isn't already there for now, but it will be after the next release and its link should be  https://docs.rs/bevy/latest/bevy/prelude/struct.WindowDescriptor.html#structfield.transparent

Comment (louisgjohnson) [reply examples/window/transparent_window.rs]: updated

File: crates/bevy_window/src/window.rs [hunk 2]
```diff
@@ -535,6 +535,13 @@ pub struct WindowDescriptor {
     pub cursor_visible: bool,
     pub cursor_locked: bool,
     pub mode: WindowMode,
+    /// Sets whether the background of the window should be transparent.
+    /// # Platform-specific
+    /// - iOS / Android / Web: Unsupported.
+    /// - OSX / Linux : Not working as expected.
+    /// OSX transparent works with winit out of the box, so this issue might be related to: https://github.com/gfx-rs/wgpu/issues/687
+    /// Linux now works with this pr merged in but bevy is using an older version of winit: https://github.com/rust-windowing/winit/pull/2006
```
Comment (mockersf) [inline crates/bevy_window/src/window.rs:543]: Bevy is using the latest released version of Winit, there hasn't been a release since this PR was merged

Comment (louisgjohnson) [reply crates/bevy_window/src/window.rs:543]: Sorry, I should of said master branch of winit, my mistake. I'll update the comment now

**Review by alice-i-cecile — APPROVED** [review]

**Review by mockersf — APPROVED** [review]

**Comment by rezural** [comment]
@louisgjohnson I can confirm this works on Ubuntu 20.04 with nvidia 1070, on bevy main.
